jib-core/src/main/java/com/google/cloud/tools/jib/filesystem/PermissionsHelper.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2018 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.filesystem;
+
+import com.google.common.base.Preconditions;
+import java.nio.file.attribute.PosixFilePermission;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+
+/** Helpers for converting file permissions. */
+public class PermissionsHelper {
+
+  private static final Map<PosixFilePermission, Integer> permissionMap;
```

Comment:
An alternative for this would be to just create an `ImmutableMap` using `ImmutableMap.of` directly.

Reply:
I think `ImmutableMap.of` takes at most 5 entries.

Reply:
Ah okay, then it could be `ImmutableMap.builder().put(...).put(...)....build()`